tests/puppeteer/pages/BO/brands.js

```diff
@@ -0,0 +1,303 @@
+      default:
+      // Do nothing
+    }
+    // click on search
+    await this.clickAndWaitForNavigation(searchButton);
+  }
+
+  /**
+   * Filter Brands
+   * @param filterType, input / Select
+   * @param filterBy, which column
+   * @param value, value to put in filter
+   * @return {Promise<void>}
+   */
+  async filterBrands(filterType, filterBy, value = '') {
+    await this.filterTable('manufacturer', filterType, filterBy, value);
+  }
+
+  /**
+   * Filter Addresses
+   * @param filterType, input / Select
+   * @param filterBy, which column
+   * @param value, value to put in filter
+   * @return {Promise<void>}
+   */
+  async filterAddresses(filterType, filterBy, value = '') {
+    await this.filterTable('manufacturer_address', filterType, filterBy, value);
+  }
+
+  /**
+   * Get toggle column value for a row (Brands list)
+   * @param row
+   * @return {Promise<string>}
+   */
+  async getToggleColumnValue(row) {
+    return this.elementVisible(
+      this.brandsEnableColumnValidIcon.replace('%ROW', row), 100);
+  }
+
+  /**
+   * Update Enable column for the value wanted in Brands list
+   * @param row
+   * @param valueWanted
+   * @return {Promise<boolean>}, true if click has been performed
+   */
+  async updateEnabledValue(row, valueWanted = true) {
+    if (await this.getToggleColumnValue(row, 'active') !== valueWanted) {
+      await this.clickAndWaitForNavigation(this.brandsTableEnableColumn.replace('%ROW', row));
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Go to New Brand Page
+   * @return {Promise<void>}
+   */
+  async goToAddNewBrandPage() {
+    await this.clickAndWaitForNavigation(this.newBrandLink);
+  }
+
+  /**
+   * Go to new Brand Address Page
+   * @return {Promise<void>}
+   */
+  async goToAddNewBrandAddressPage() {
+    await this.clickAndWaitForNavigation(this.newBrandAddressLink);
+  }
+
+  /**
+   * View Brand
+   * @param row, Which row of the list
+   * @return {Promise<void>}
+   */
+  async viewBrand(row = '1') {
+    await this.clickAndWaitForNavigation(this.viewBrandLink.replace('%ROW', row));
+  }
+
+  async goToEditBrandPage(row = '1') {
+    await Promise.all([
+      this.page.click(this.dropdownToggleButton.replace('%TABLE', 'manufacturer').replace('%ROW', row)),
+      this.page.waitForSelector(
+        `${this.dropdownToggleButton}[aria-expanded='true']`
+          .replace('%TABLE', 'manufacturer').replace('%ROW', row),
+      ),
+    ]);
+    await this.clickAndWaitForNavigation(this.editBrandLink.replace('%ROW', row));
+  }
+
+  /**
+   *
+   * @param row
+   * @return {Promise<void>}
+   */
+  async goToEditBrandAddressPage(row = '1') {
+    await this.clickAndWaitForNavigation(this.editBrandAddressLink.replace('%ROW', row));
+  }
+
+  /**
+   * Delete Row in table
+   * @param table, brand or address
+   * @param row, row to delete
+   * @return {Promise<textContent>}
+   */
+  async deleteRowInTable(table, row = '1') {
+    this.dialogListener(true);
```

Comment:
What is this for ?

Reply:
To accept javascript dialog with message "Delete Selected items"
![image](https://user-images.githubusercontent.com/48441421/67579903-ac7a2180-f745-11e9-82cd-df8541f48c9f.png)